cli/program_state.rs

```diff
@@ -279,261 +289,66 @@ impl ProgramState {
   }
 }
 
-/// Determine if TS compiler should be run with `allowJs` setting on. This
-/// is the case when there's either:
-///  - a JavaScript file with non-JavaScript import
-///  - JSX import
-fn should_allow_js(module_graph_files: &[&ModuleGraphFile]) -> bool {
-  module_graph_files.iter().any(|module_file| {
-    if module_file.media_type == MediaType::JSX {
-      true
-    } else if module_file.media_type == MediaType::JavaScript {
-      module_file.imports.iter().any(|import_desc| {
-        let import_file = module_graph_files
-          .iter()
-          .find(|f| {
-            f.specifier == import_desc.resolved_specifier.to_string().as_str()
-          })
-          .expect("Failed to find imported file");
-        let media_type = import_file.media_type;
-        media_type == MediaType::TypeScript
-          || media_type == MediaType::TSX
-          || media_type == MediaType::JSX
-      })
+// TODO(@kitsonk) this is only temporary, but should be refactored to somewhere
+// else, like a refactored file_fetcher.
+impl SourceMapGetter for ProgramState {
```

Comment:
Glad to see `SourceMapGetter` moved from `tsc.rs` - I do believe it should be implemented on `ProgramState`

Reply:
Well, I think it should be in what ever file_fetcher becomes... whatever "owns" caching the emitted source files/source maps.  But obviously something for later.